File: tests/fixtures/vulnerable.java
// Java test fixture — intentionally vulnerable code for foxguard detection tests

import java.io.*;
import java.net.URL;
import java.sql.*;
import javax.xml.parsers.DocumentBuilderFactory;

public class Vulnerable {

    // java/no-sql-injection
    void sqlInjection(String userId) throws Exception {
        Statement stmt = conn.createStatement();
        stmt.executeQuery("SELECT * FROM users WHERE id = " + userId);
        em.createQuery("SELECT u FROM User u WHERE u.name = '" + name + "'");
    }

    // java/no-command-injection
    void commandInjection(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
        new ProcessBuilder(cmd);
    }

    // java/no-unsafe-deserialization
    void unsafeDeser(InputStream is) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(is);
        ois.readObject();
    }

    // java/no-ssrf
    void ssrf(String url) throws Exception {
        new URL(url);
    }

    // java/no-path-traversal
    void pathTraversal(String path) throws Exception {
        new File(path);
        new FileInputStream(path);
    }

    // java/no-weak-crypto
    void weakCrypto() throws Exception {
        MessageDigest.getInstance("MD5");
        Cipher.getInstance("DES");
        Cipher.getInstance("AES/ECB/PKCS5Padding");
    }

    // java/no-hardcoded-secret
    String password = "supersecret123";
    String apiKey = "sk-live-abcdef123456";

    // java/no-xxe
    void xxe() throws Exception {
        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
    }

    // java/spring-csrf-disabled
    void csrf() {
        http.csrf().disable();
    }

    // java/spring-cors-permissive
    void cors() {
        registry.allowedOrigins("*");
    }

    // java/no-xss
    void xss(String userInput, HttpServletResponse response) throws Exception {
        response.getWriter().write(userInput);
        response.getWriter().println(userInput);
        out.write(userInput);
        out.println(userInput);
        PrintWriter pw = response.getWriter();
        pw.write("<h1>" + userInput);
        pw.println("<div>" + userInput);
    }
}

File: tests/fixtures/vulnerable.kt
import java.io.File
import java.net.URL

// Kotlin test fixture — intentionally vulnerable code for foxguard detection tests.

fun sqlInjection(db: Database, userId: String) {
    db.executeQuery("SELECT * FROM users WHERE id = " + userId)
}

fun commandInjection(command: String) {
    Runtime.getRuntime().exec(command)
    ProcessBuilder(command).start()
}

fun unsafeDeserialization(input: java.io.InputStream) {
    val stream = java.io.ObjectInputStream(input)
    stream.readObject()
}

fun ssrf(url: String) {
    URL(url).readText()
}

fun pathTraversal(path: String) {
    File(path).readText()
}

fun weakCrypto(data: ByteArray) {
    java.security.MessageDigest.getInstance("MD5").digest(data)
}

fun hardcodedSecret() {
    val apiKey = "sk-live-abcdef123456789"
    println(apiKey)
}

fun xxe(xml: String) {
    val factory = javax.xml.parsers.DocumentBuilderFactory.newInstance()
    val builder = factory.newDocumentBuilder()
    builder.parse(xml.byteInputStream())
}

fun cors(config: CorsRegistry) {
    config.addMapping("/**").allowedOrigins("*")
}

fun eval(script: String) {
    javax.script.ScriptEngineManager().getEngineByName("js").eval(script)
}

fun taintSql(call: ApplicationCall, db: Database) {
    val id = call.request.queryParameters["id"]
    db.executeQuery("SELECT * FROM users WHERE id = " + id)
}
